uktrade/great-cms
# Event listing page frontend

This PR adds the frontend styling and functionality to the event listing page.

In order to build to the design provided, changes include:
- A new `HeroSnippet` model, which will allow the hero for the page to be editable in wagtail
- A reusable 'show-more' html component which can be used to truncate and hide text
- A new `eventFilters.js` file. This repeats logic used in `marketFilters.js` which I recognise is inefficient. Following this PR I will do a refactor to see what logic can be shared between the two sets of filters, and make it easier for future filters to be used.

**To test**
- Apply latest migrations
- Add a snippet for the hero (page will not load without it) [here in wagtail](http://greatcms.trade.great:8020/admin/snippets/core/herosnippet/)
- Go to http://greatcms.trade.great:8020/export-academy/upcoming-events/
- With javascript turned on, selecting the checkboxes/radio buttons should apply filters
- With javascript turned off, an 'Update results' button should appear below the filter which will apply filters

**Next steps**
- Develop the mobile view (awaiting design being signed off)
- Refactor filters javascript logic to be reusable
- Add text for when no events match the filters (awaiting design)

**With javascript on**
![Screenshot 2023-03-21 at 16 27 17](https://user-images.githubusercontent.com/22460823/226679004-cf797979-870d-4ddd-9697-aeb3457bab22.png)

**Pagination**
![Screenshot 2023-03-21 at 16 26 45](https://user-images.githubusercontent.com/22460823/226679156-a441db74-c9a9-4488-83ef-465ed49120a5.png)

**With javascript off**
![Screenshot 2023-03-21 at 16 37 50](https://user-images.githubusercontent.com/22460823/226679428-a91a6e82-d42f-4048-9132-95d1814dc212.png)


### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/KLS-269

### Reviewing help


### Housekeeping


### Merging
